### PR TITLE
Add [StringSyntax("Route")] to route patterns

### LINKED ISSUE #2148: Add `[StringSyntax("Route")]` to route patterns
ASP.NET Core adds syntax highlighting and various other features to route patterns. This feature is enabled by adding an attribute to method parameters:

https://github.com/dotnet/aspnetcore/blob/f56c242d037e7a11626151e78333caa26c3b5d00/src/Middleware/HealthChecks/src/Builder/HealthCheckEndpointRouteBuilderExtensions.cs#L28

YARP should add this attribute to places it has route patterns. For example:

https://github.com/microsoft/reverse-proxy/blob/5529aca33859db3dcac2b6a68562b47b5d6bb2a3/src/ReverseProxy/Routing/DirectForwardingIEndpointRouteBuilderExtensions.cs#L22

### PR DESCRIPTION
**Pull request by divyeshio**
Fixes #2148 

Adds [StringSyntax("Route")] to route patterns

### CODE REVIEW

**Comment by divyeshio** [comment]
> @divyeshio please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

File: src/ReverseProxy/Routing/DirectForwardingIEndpointRouteBuilderExtensions.cs [hunk 1]
```diff
@@ -63,15 +84,19 @@ public static IEndpointConventionBuilder MapForwarder(this IEndpointRouteBuilder
     /// <summary>
     /// Adds direct forwarding of HTTP requests that match the specified pattern to a specific destination using customized configuration for the outgoing request, default transforms, and default HTTP client.
     /// </summary>
-    public static IEndpointConventionBuilder MapForwarder(this IEndpointRouteBuilder endpoints, string pattern, string destinationPrefix, ForwarderRequestConfig requestConfig)
+    public static IEndpointConventionBuilder MapForwarder(this IEndpointRouteBuilder endpoints, [StringSyntax("Route")] string pattern, string destinationPrefix, ForwarderRequestConfig requestConfig)
```
Comment (Tratcher) [inline src/ReverseProxy/Routing/DirectForwardingIEndpointRouteBuilderExtensions.cs:87]: Missed a #if

Comment (Tratcher) [reply src/ReverseProxy/Routing/DirectForwardingIEndpointRouteBuilderExtensions.cs:87]:
```suggestion
    public static IEndpointConventionBuilder MapForwarder(this IEndpointRouteBuilder endpoints,
    #if NET7_0_OR_GREATER
    [StringSyntax("Route")] string pattern,
    #endif
    string destinationPrefix, ForwarderRequestConfig requestConfig)
```

**Review by Tratcher — APPROVED** [review]

**Comment by divyeshio** [comment]
Do you know reason why CI is still failing?

File: src/ReverseProxy/Routing/DirectForwardingIEndpointRouteBuilderExtensions.cs [hunk 2]
```diff
@@ -63,15 +84,23 @@ public static IEndpointConventionBuilder MapForwarder(this IEndpointRouteBuilder
     /// <summary>
     /// Adds direct forwarding of HTTP requests that match the specified pattern to a specific destination using customized configuration for the outgoing request, default transforms, and default HTTP client.
     /// </summary>
-    public static IEndpointConventionBuilder MapForwarder(this IEndpointRouteBuilder endpoints, string pattern, string destinationPrefix, ForwarderRequestConfig requestConfig)
+    public static IEndpointConventionBuilder MapForwarder(this IEndpointRouteBuilder endpoints,
+    #if NET7_0_OR_GREATER
+    [StringSyntax("Route")] string pattern,
+    #endif
+    string destinationPrefix, ForwarderRequestConfig requestConfig)
```
Comment (Tratcher) [inline src/ReverseProxy/Routing/DirectForwardingIEndpointRouteBuilderExtensions.cs:91]:
```suggestion
    #if NET7_0_OR_GREATER
    [StringSyntax("Route")]
    #endif
     string pattern, string destinationPrefix, ForwarderRequestConfig requestConfig)
```

**Comment by Tratcher** [comment]
It was a bad edit on my part. Should be fixed now.

**Pull request merged**